src/gwt/src/org/rstudio/studio/client/workbench/addins/AddinsCommandManager.java

```diff
@@ -72,7 +73,7 @@ public void onAddinsKeyBindingsChanged(AddinsKeyBindingsChangedEvent event)
                @Override
                public void onAddinRegistryUpdated(AddinRegistryUpdatedEvent event)
                {
-                  rAddins_ = event.getData();
+                  MainWindowObject.rAddins().set(event.getData());
```

Comment:
I think you're right, as the `AddinRegistryUpdatedEvent` is fired by the server, and all windows will receive that event. I'll tidy that up -- since the main window will always receive this event it's okay for the main window to be in charge of setting the field.

Nice catch!